### PR TITLE
Add NotInSet operator

### PR DESCRIPTION
**Pull request by awarrenlove**
Add a NotInSet operator to be the inverse of InSet. This also updates the serializer to check for exact class matches so Operators can extend each other without breaking serialization.

### CODE REVIEW

File: test/Qandidate/Toggle/Operator/NotInSetTest.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            array(2,     array(1, 1, 2, 3)),
+            array('qux', array('qux', 'bar')),
+        );
+    }
+
+    /**
+     * @test
+     * @dataProvider nullSets
+     */
+    public function it_always_accept_NULL_as_part_of_a_set($argument, $set)
```
Comment (othillo) [inline test/Qandidate/Toggle/Operator/NotInSetTest.php]: for clarity shouldn't we rename this to it_considers_NULL_to_not_be_part_of_a_set()?

Comment (awarrenlove) [reply test/Qandidate/Toggle/Operator/NotInSetTest.php]: Good call. I'll update this.

**Comment by othillo** [comment]
LGTM! can you remove the .idea directory?

**Comment by wjzijderveld** [comment]
Thanks @awarrenlove !

**Pull request merged**